### PR TITLE
Split device section into storage and network

### PR DESCRIPTION
**Pull request by edlerd**
* split device section of profile/instance create/edit forms into storage and network

### CODE REVIEW

**Comment by al-bakalova** [comment]
A couple of small things to improve:
- The title of the second section should be in plural - "Networks"
- Once the user lands on "Networks" - all the network fields should be available, with a button "Add network" at the bottom to indicate to the user that they can add more than one networks.
- The "Delete device" button label is to be changed to "Delete"
- The Delete should be visible once the user actually adds a network and not when the fields are empty

**Review by lorumic — COMMENTED** [review]
In addition to the other few minor comments, I wonder if we need the `DevicesForm` FC anymore? I would remove it and maybe move the shared types and const to a dedicated helper file.

File: static/js/pages/instances/forms/StorageForm.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+const StorageForm: FC<Props> = ({ formik, project, children }) => {
+  const { data: storagePools = [] } = useQuery({
+    queryKey: [queryKeys.storage],
+    queryFn: () => fetchStorages(project),
+  });
+
+  const index = formik.values.devices.findIndex(
+    (item) => item.type === "disk" && item.name === "root"
+  );
+  const hasStorage = index !== -1;
```
[the remark this answers is not in the document]
Comment (edlerd) [reply static/js/pages/instances/forms/StorageForm.tsx]: renamed it.

File: static/js/pages/instances/forms/StorageForm.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+  const addStorage = () => {
```
[the remark this answers is not in the document]
Comment (edlerd) [reply static/js/pages/instances/forms/StorageForm.tsx]: renamed to addRootStorage

File: static/js/pages/instances/forms/NetworkForm.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      value: "",
+    });
+    return options;
+  };
+
+  return (
+    <div className="device-form">
+      {children}
+      {formik.values.devices.map(
+        (variable, index) =>
```
[the remark this answers is not in the document]
Comment (edlerd) [reply static/js/pages/instances/forms/NetworkForm.tsx]: renamed it to `formDevice`

**Review by lorumic — APPROVED** [review]

**Pull request merged**